src/wallet/rpcwallet.cpp

```diff
@@ -671,11 +671,22 @@ static CAmount GetReceived(const CWallet& wallet, const UniValue& params, bool b
     if (!params[1].isNull())
         min_depth = params[1].get_int();
 
+    bool include_immature{!params[2].isNull() && params[2].get_bool()};
+
+    // Filtering out coinbase outputs is deprecated
+    // It can be enabled by setting deprecatedrpc=filter_coinbase
+    bool filter_coinbase{wallet.chain().rpcEnableDeprecated("filter_coinbase")};
+
     // Tally
     CAmount amount = 0;
     for (const std::pair<const uint256, CWalletTx>& wtx_pair : wallet.mapWallet) {
         const CWalletTx& wtx = wtx_pair.second;
-        if (wtx.IsCoinBase() || !wallet.chain().checkFinalTx(*wtx.tx) || wtx.GetDepthInMainChain() < min_depth) {
+        int depth{wtx.GetDepthInMainChain()};
+        if (depth < min_depth
+            // Coinbase with less than 1 confirmation is an orphan
```

Comment:
"orphan" is a very overloaded word, and I don't think it's correct here. I think you're trying to say that the coinbase transaction has been conflicted out of the main chain?